File: Array & Hashing/Is Anagram/Anagram.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        
        Map<Character, Integer> countMap = new HashMap<>();
        
        for (char c : s.toCharArray()) {
            countMap.put(c, countMap.getOrDefault(c, 0) + 1);
        }
        
        for (char c : t.toCharArray()) {
            if (!countMap.containsKey(c) || countMap.get(c) == 0) return false;
            countMap.put(c, countMap.get(c) - 1);
        }

        return true;
    }
}


public class Anagram {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Solution obj = new Solution();
        int tc = sc.nextInt();
        sc.nextLine();
        while(tc > 0) {
            String s = sc.nextLine();
            String t = sc.nextLine();
            System.out.println(obj.isAnagram(s, t));
            tc--;
        }
        sc.close();
    }
}

File: Array & Hashing/Is Anagram/Anagram.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
 public:
  bool isAnagram(const string &s, const string &t) {
    if (s.length() != t.length()) return false;
    unordered_map<char, int> countMap;
    for (char c : s) {
      countMap[c]++;
    }
    for (char c : t) {
      if (countMap.find(c) == countMap.end() || countMap[c] == 0) return false;
      countMap[c]--;
    }
    return true;
  }
};

int main() {
  Solution obj;
  int tc;
  cin >> tc;
  cin.ignore();
  while (tc-- > 0) {
    string s, t;
    getline(cin, s);
    getline(cin, t);
    cout << (obj.isAnagram(s, t) ? "true" : "false") << endl;
  }
  return 0;
}